hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -679,32 +679,60 @@ public AuthType getAuthType() {
 
   @VisibleForTesting
   String initializeUserAgent(final AbfsConfiguration abfsConfiguration,
-                             final String sslProviderName) {
+      final String sslProviderName) {
```

Comment:
if you pulled this out somewhere or made static you could have a unit test which would validate some of this

Reply:
There are test cases written to validate this as this method is package protected and with the @VisibleForTesting.